mobile_verifier/src/boosting_oracles/data_sets.rs

```diff
@@ -0,0 +1,492 @@
+            .await?;
+            tracing::info!(
+                data_set = latest_unprocessed_data_set.filename,
+                "Data set download complete"
+            );
+        }
+
+        // Now that we've downloaded the file, load it into the data set
+        data_set.update(Path::new(&path), latest_unprocessed_data_set.time_to_use)?;
+
+        // Release the lock as it is no longer needed
+        drop(data_set);
+
+        // Update the hexes
+        let boosting_reports = set_oracle_boosting_assignments(
+            UnassignedHex::fetch_all(&self.pool),
+            &self.data_sets,
+            &self.pool,
+        )
+        .await?;
+
+        db::set_data_set_status(
+            &self.pool,
+            &latest_unprocessed_data_set.filename,
+            DataSetStatus::Processed,
+        )
+        .await?;
+
+        self.oracle_boosting_sink
+            .write_all(boosting_reports)
+            .await?;
+        self.oracle_boosting_sink.commit().await?;
+        tracing::info!(
+            data_set = latest_unprocessed_data_set.filename,
+            "Data set processing complete"
+        );
+
+        Ok(())
+    }
+
+    pub async fn run(mut self) -> anyhow::Result<()> {
+        let poll_duration = Duration::minutes(5);
+
+        loop {
+            self.check_for_available_data_sets().await?;
+            self.process_data_sets().await?;
+            tokio::time::sleep(poll_duration.to_std()?).await;
+        }
+    }
+}
+
+fn get_data_set_path(
+    data_set_directory: &Path,
+    data_set_type: DataSetType,
+    time_to_use: DateTime<Utc>,
+) -> PathBuf {
+    let path = PathBuf::from(format!(
+        "{}.{}.res10.h3tree",
+        data_set_type.to_prefix(),
+        time_to_use.timestamp()
+    ));
+    let mut dir = data_set_directory.to_path_buf();
+    dir.push(path);
+    dir
+}
+
+async fn download_data_set(
+    store: &FileStore,
+    in_file_name: &str,
+    out_path: &Path,
+) -> anyhow::Result<()> {
+    tracing::info!("Downloading new data set: {}", out_path.to_string_lossy());
+    let stream = store.get_raw(in_file_name).await?;
+    let mut bytes = tokio_util::codec::FramedRead::new(
+        async_compression::tokio::bufread::GzipDecoder::new(tokio_util::io::StreamReader::new(
+            stream,
+        )),
+        tokio_util::codec::BytesCodec::new(),
+    );
+    let mut file = File::create(&out_path).await?;
+    while let Some(bytes) = bytes.next().await.transpose()? {
+        file.write_all(&bytes).await?;
+    }
+    Ok(())
+}
+
+#[derive(Copy, Clone, sqlx::Type)]
+#[sqlx(type_name = "data_set_type")]
+#[sqlx(rename_all = "lowercase")]
+pub enum DataSetType {
+    Urbanization,
+    Footfall,
+    Landtype,
+}
+
+impl DataSetType {
+    pub fn to_prefix(self) -> &'static str {
+        match self {
+            Self::Urbanization => "urbanization",
+            Self::Footfall => "footfall",
+            Self::Landtype => "landtype",
+        }
+    }
+}
+
+pub mod db {
```

Comment:
The migration missing from this module. 

Considering the purpose of the table, perhaps a more descriptive name than 'data_sets' could provide clearer context.

Reply:
`The migration missing from this module.`
what does this mean?